protocol/examples/messageData/election_setup/bad_election_setup_question_identical_id.json

```diff
@@ -0,0 +1,27 @@
+{
+  "object": "election",
+  "action": "setup",
+  "version": "OPEN_BALLOT",
+  "id": "zG1olgFZwA0m3mLyUqeOqrG0MbjtfqShkyZ6hlyx1tg=",
+  "lao": "fzJSZjKf-2cbXH7kds9H8NORuuFIRLkevJlN7qQemjo=",
+  "name": "Election",
+  "created_at": 1633098941,
+  "start_time": 1633098941,
+  "end_time": 1633099812,
+  "questions": [
+    {
+      "id": "2PLwVvqxMqW5hQJXkFpNCvBI9MZwuN8rf66V1hS-iZU=",
+      "question": "Is this project fun?",
+      "voting_method": "Plurality",
+      "ballot_options": ["Yes", "No"],
+      "write_in": false
+    },
+    {
+      "id": "2PLwVvqxMqW5hQJXkFpNCvBI9MZwuN8rf66V1hS-iZU=",
+      "question": "Is this project fun?",
+      "voting_method": "Plurality",
+      "ballot_options": ["Yes", "Yes"],
+      "write_in": true
+    }
+  ]
+}
```

Comment:
Great that you saw the protocol folder! 😁

Usually in addition to the examples there is a test file that makes sure that bad examples do not pass the json schema validation and good examples do. You can find it at https://github.com/dedis/popstellar/blob/master/protocol/test/main.test.js. But to be fair I don't think this is something that can be checked with JSON schemas by default. Although it seems the validator we're using (at least in protocol tests and also on fe-1) has a keyword https://github.com/ajv-validator/ajv-keywords#uniqueitemproperties that can be used for this :)

Moreover it would be good to document the change in https://github.com/dedis/popstellar/blob/master/docs/messageData.md#setting-up-an-election-electionsetup.

Reply:
Thank you for pointing that out ! I added the example in main.test.js. It looks like this case can be checked since https://github.com/dedis/popstellar/blob/master/docs/messageData.md#setting-up-an-election-electionsetup already specifies "uniqueItems" as true. Considering this, I think the docs is already complete (probably thanks to fe1 because I noticed that this bug was fixed from their end). How does it look now ?

Reply:
Thank you for adding it! :)

I just double checked since `uniqueItems` should not check the individual properties of the array elements but just the whole objects and it turns out that the test passed (i.e. validation failed) because of other issues with the file that I missed before. I added comments in the respective places.

Hence I'd still update the docs a bit to make sure that the "question" property has to be unique. You can use the "$comment" field for this.

Also I quickly checked out `uniqueitemproperties` that I linked in my previous comment and it seems there is only partial support for it and since it might lead to other systems having issues with a non-standard key (genuinely don't know but I think if we want to enforce it, it should be enforced on all subsystems). I would suggest to either forget about this and make sure the documentation & implementation make clear that the question property has to be unique OR look for some other solution?

Also I just reviewed the changes regarding the protocol, not the changes in the go code. You will need to ask @Anghul for a review of it 😄

Reply:
Thank you for the feedback ! I just pushed the changes you suggested: updating the docs and just commenting the test as one that cannot be checked by json schemas like some other ones above it.